contentcuration/contentcuration/static/js/edit_channel/uploader/components/DialogBox/DialogBox.vue

```diff
@@ -0,0 +1,45 @@
+<template>
```

Comment:
Oh yes I noticed but forgot to mention, I'm sorry about that - I was thinking about merging dialog components in another PR when working on unification of our work and chatting about it at first with you because it's something that will also affect our future usage of the dialog from other parts of the app. It also relates to your comment about `v-model` so I will try to give summary here and later we can decide if it's something that we could use or needs to be adjusted. I didn't want to refactor stuff to soon because I don't have a very good knowledge of our previous use cases. Mechanism behind this dialog architecture is customized to following:

I need to open it from more pure UI components located deeper in the components tree (`AssessmentEditor`, `AssessmentItemEditor`) and at the same time I don't want to include the dialog component within each one of them for following reasons:
1. In ideal case it should be placed at one place in DOM only (temporarily it's `AssessmentView` but I think we should go with something higher in the Studio views structure) and controlled by `v-show`
2. Based on discussion with Richard, `AssessmentEditor` should be decoupled from Studio in general. If we'll use it in another app, it will probably also use another dialog because it's usually something global and specific to the application
3. I guess we might have more and more similar use cases so it would be nice to not need to include dialog component in smaller components.

=> currently you can provide optional `openDialog` function with a certain attributes interface to `AssessmentEditor` and `AssessmentItemEditor`. If it's not provided, no confirmation will be needed. If it's provided, those components will use the function. This function can be then connected to store (or whatever is needed) that holds information if dialog is open and from that it can propagate to dialog component placed somewhere high in the views structure. This could be combined with `v-model`. If I can remember well, I think that `DialogBox` interface is built on `v-model`.

Reply:
Thank you for clarifying! That makes sense to me (and your approach would really simplify some of the edit modal templates). Let's definitely revisit after the refactor!